nixos/modules/services/matrix/synapse.nix

```
@@ -158,6 +170,59 @@ in {
         '';
       };
 
+      extras = mkOption {
+        type = types.listOf (types.enum (lib.attrNames cfg.package.optional-dependencies));
+        default = [
+          "postgres"
+          "systemd"
+          "url-preview"
+          "user-search"
+        ];
+        example = literalExpression ''
+          [
+            # Provide statistics about caching memory consumption
+            "cache-memory"
+
+            # JSON Web Token authentication
+            "jwt"
+
+            # End-to-end tracing support using Jaeger
+            "opentracing"
+
+            # OpenID Connect authentication
+            "oidc"
+
+            # PostgreSQL database backend
+            "postgresql"
+
+            # Redis support for the replication stream between worker processes
+            "redis"
+
+            # SAML2 authentication
+            "saml2"
+
+            # Error tracking and performance metrics
+            "sentry"
+
+            # Provide the JournalHandler used in the default log_config
+            "systemd"
+
+            # Support for oEmbed URL previews
+            "url-preview"
+
+            # Support internationalized domain names in user-search
+            "user-search"
+          ]
```

```suggestion
          [
            "cache-memory" # Provide statistics about caching memory consumption
            "jwt"          # JSON Web Token authentication
            "opentracing"  # End-to-end tracing support using Jaeger
            "oidc"         # OpenID Connect authentication
            "postgresql"   # PostgreSQL database backend
            "redis"        # Redis support for the replication stream between worker processes
            "saml2"        # SAML2 authentication
            "sentry"       # Error tracking and performance metrics
            "systemd"      # Provide the JournalHandler used in the default log_config
            "url-preview"  # Support for oEmbed URL previews
            "user-search"  # Support internationalized domain names in user-search
          ]
```
we should condense that a bit otherwise it will probably poorly display on search.nixos.org.

---

I'm not sure how it is going to look on s.n.o.. Do you have other examples in mind, which we can use to get an idea?

---

``services.nextcloud.extraApps`` has a longer example, `services.tor.relay.role` has a longer description.